openmls/src/treesync/node/leaf_node.rs

```diff
@@ -545,26 +551,39 @@ impl LeafNode {
         }
     }
 
-    /// Returns `true` if the [`ExtensionType`] is supported by this leaf node.
-    pub(crate) fn supports_extension(&self, extension_type: &ExtensionType) -> bool {
-        self.payload
-            .capabilities
-            .extensions
-            .iter()
-            .any(|et| et == extension_type)
-            || default_extensions().iter().any(|et| et == extension_type)
+    /// Returns the [`Lifetime`] if present.
+    /// `None` otherwise.
+    pub(crate) fn life_time(&self) -> Option<&Lifetime> {
+        if let LeafNodeSource::KeyPackage(life_time) = &self.payload.leaf_node_source {
+            Some(life_time)
+        } else {
+            None
+        }
     }
 
-    /// Returns `true` if the [`ProposalType`] is supported by this leaf node.
-    pub(crate) fn supports_proposal(&self, proposal_type: &ProposalType) -> bool {
-        self.payload
-            .capabilities
-            .proposals
-            .iter()
-            .any(|pt| pt == proposal_type)
-            || default_proposals().iter().any(|pt| pt == proposal_type)
+    /// Returns a reference to the [`Signature`] of this leaf.
+    pub fn signature(&self) -> &Signature {
+        &self.signature
+    }
+
+    /// Return a reference to [`Capabilities`].
+    pub(crate) fn capabilities(&self) -> &Capabilities {
+        &self.payload.capabilities
+    }
+
+    /// Return a mutable reference to [`Capabilities`].
+    #[cfg(test)]
+    pub fn capabilities_mut(&mut self) -> &mut Capabilities {
```

Comment:
This could be moved to a different `impl` as well. If not, let's at least put it at the bottom of the getters or something.

Reply:
Is the difference between `#[cfg(any(feature = "test-utils", test))]` and ` #[cfg(test)]` relevant? Can we unify that into the former?

Reply:
You will probably get dead code warnings because the function is not used outside of unit tests.

Reply:
Right... I was thinking about making it public through the `test-utils` feature. But not all methods are currently well-suited for this as they would leak private types. Let's keep this annotations as they are for now.